packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,374 @@
+// Disabling because returnTypes must be last param to match 1.x params
+/* eslint-disable default-param-last */
+
+import { TransactionCall, TransactionWithSender, Web3BaseProvider } from 'web3-common';
+import { Web3Config, Web3RequestManager } from 'web3-core';
+import {
+	BlockNumberOrTag,
+	ValidTypes,
+	ValidReturnTypes,
+	convertToValidType,
+	Address,
+	Uint256,
+	HexString32Bytes,
+	HexStringBytes,
+	Uint,
+	HexString8Bytes,
+	convertObjectPropertiesToValidType,
+	isHexString32Bytes,
+	Filter,
+} from 'web3-utils';
+import {
+	convertibleBlockProperties,
+	convertibleFeeHistoryResultProperties,
+	convertibleReceiptInfoProperties,
+	convertibleTransactionInfoProperties,
+} from './convertible_properties';
+
+import * as rpcMethods from './rpc_methods';
+
+export class Web3Eth {
+	private readonly _requestManager: Web3RequestManager;
+	private readonly _options: Web3Config;
+
+	public constructor(provider: Web3BaseProvider | string, options: Web3Config) {
+		this._requestManager = new Web3RequestManager(provider);
+		this._options = options;
+	}
+
+	public async getProtocolVersion() {
```

Comment:
we must specify return type of this and all following functions.

Reply:
Typescript automatically infers the return type, no need to explicitly declare it